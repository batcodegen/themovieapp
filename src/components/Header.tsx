import {StyleSheet, Text, View} from 'react-native';
import React from 'react';
import Ionicons from 'react-native-vector-icons/Ionicons';
import {DrawerActions, useNavigation, useTheme} from '@react-navigation/native';
import {Pressable} from 'react-native';

interface HeaderProps {
  screenName: string;
}

const Header = ({screenName}: HeaderProps) => {
  const {colors} = useTheme();
  const navigation = useNavigation();
  return (
    <View style={styles.container}>
      <Pressable
        style={styles.iconContainer}
        onPress={() => navigation.dispatch(DrawerActions.openDrawer())}>
        <Ionicons name="menu" size={30} color={colors.text} />
      </Pressable>
      <View style={styles.titleContainer}>
        <Text style={[styles.titleText, {color: colors.text}]}>
          {screenName}
        </Text>
      </View>
      <View style={styles.rightContainer} />
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    height: 50,
    alignItems: 'center',
    flexDirection: 'row',
    paddingHorizontal: 10,
  },
  iconContainer: {flex: 0.2, alignItems: 'flex-start'},
  titleContainer: {flex: 0.6, alignItems: 'center'},
  titleText: {fontSize: 18, fontWeight: 'bold', letterSpacing: 0.2},
  rightContainer: {flex: 0.2},
});
