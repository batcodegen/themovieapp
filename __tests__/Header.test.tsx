import React from 'react';
import {render, screen} from '@testing-library/react-native';
import Header from '../src/components/Header';

describe('Header Component', () => {
  it('Renders correctly', () => {
    render(<Header screenName="Popular Movies" />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
