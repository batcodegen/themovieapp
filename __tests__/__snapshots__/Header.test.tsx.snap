// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 50,
      "paddingHorizontal": 10,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "flex-start",
        "flex": 0.2,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 30,
          },
          undefined,
          {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      îµ’
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 0.6,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "letterSpacing": 0.2,
          },
          {
            "color": undefined,
          },
        ]
      }
    >
      Popular Movies
    </Text>
  </View>
  <View
    style={
      {
        "flex": 0.2,
      }
    }
  />
</View>
`;
